src/bin/tool.rs

```diff
@@ -0,0 +1,293 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::Context;
+use clap::Parser;
+use ethers::core::rand::{thread_rng, RngCore};
+use semaphore::identity::Identity;
+use semaphore::poseidon_tree::Proof;
+use semaphore::Field;
+use serde::{Deserialize, Serialize};
+use signup_sequencer::server::data::{
+    InclusionProofRequest, InclusionProofResponse, InsertCommitmentRequest,
+    VerifySemaphoreProofRequest, VerifySemaphoreProofResponse,
+};
+
+#[derive(Debug, Parser)]
+struct Args {
+    #[clap(subcommand)]
+    subcommand: Command,
+
+    #[clap(
+        short,
+        long,
+        env,
+        default_value = "https://signup-orb-ethereum.crypto.worldcoin.dev"
+    )]
+    sequencer_url: String,
+
+    #[clap(long, env)]
+    basic_auth_username: Option<String>,
+
+    #[clap(long, env)]
+    basic_auth_password: Option<String>,
+
+    /// The path to the file that will be used to store the identity
+    #[clap(long, env)]
+    identity_file: Option<PathBuf>,
+
+    /// The path to the file that will be used to store the inclusion proof
+    #[clap(long, env)]
+    inclusion_proof_file: Option<PathBuf>,
+
+    /// The path to the file that will be used to store the semaphore proof
+    #[clap(long, env)]
+    semaphore_proof_file: Option<PathBuf>,
+}
```

Comment:
It looks specific args are only used for certian commands. Consider using a similar [approach to forge](https://github.com/foundry-rs/foundry/blob/master/crates/forge/bin/opts.rs#L36-L173) where each command has it's own set of args.